src/components/MoneyRequestConfirmationList.js

```diff
@@ -290,6 +290,12 @@ function MoneyRequestConfirmationList(props) {
         return [...selectedParticipants, OptionsListUtils.getIOUConfirmationOptionsFromPayeePersonalDetail(payeePersonalDetails)];
     }, [selectedParticipants, props.hasMultipleParticipants, payeePersonalDetails]);
 
+    const distanceMerchant = useMemo(() => DistanceRequestUtils.getDistanceMerchant(distance, unit, rate, currency, translate), [distance, unit, rate, currency, translate]);
+
+    useEffect(() => {
+        IOU.setMoneyRequestMerchant(distanceMerchant);
```

Comment:
I don't like this pattern though.

We should get the distance request merchant string and set the string in IOU at the same place.

We can put the following code inside `useEffect` with the dependency array.

```ts
const distanceMerchant = DistanceRequestUtils.getDistanceMerchant(distance, unit, rate, currency, translate)
IOU.setMoneyRequestMerchant(distanceMerchant);
```

And for the `title` of `MenuItemWithTopDescription`, we can pass `iou.merchant` instead.

I gonna approve this PR right now, but please make a follow up PR @Santhosh-Sellavel for the cleanup 🙇 

We consider the issue to be complete after the follow-up PR is merged 👍

Reply:
@hayata-suenaga Okay.